auth/credentials/filetypes.go

```diff
@@ -124,8 +124,19 @@ func resolveUniverseDomain(optsUniverseDomain, fileUniverseDomain string) string
 }
 
 func handleServiceAccount(f *credsfile.ServiceAccountFile, opts *DetectOptions) (auth.TokenProvider, error) {
+	ud := resolveUniverseDomain(opts.UniverseDomain, f.UniverseDomain)
 	if opts.UseSelfSignedJWT {
 		return configureSelfSignedJWT(f, opts)
+	} else if ud != "" && ud != internalauth.DefaultUniverseDomain {
+		// For non-GDU universe domains, token exchange is impossible and services
+		// must support self-signed JWTs.
+		if opts.Audience == "" && len(opts.scopes()) == 0 {
+			// Audience or scopes must be provided, although the scopes may be
+			// the default scopes available in resolveDetectOptions.
+			return nil, fmt.Errorf("credentials: audience or scopes must be provided for self-signed JWT")
+		}
```

Comment:
why is this check only being run for non-GDU?

Reply:
It's possible that this is redundant. I'll check.

Reply:
I think this could be validated for both safely by moving it into `configureSelfSignedJWT`

Reply:
In fact, the check I'm adding here is redundant. I'll remove it. Good catch Slava, thanks. https://github.com/googleapis/google-cloud-go/blob/main/auth/credentials/detect.go#L168

Reply:
Correction, the check linked above is to error is BOTH scope and aud are set. I believe no check is needed if neither is set, the error should be raised by the attempt to sign.